docs/howto/determining-packages.md

```diff
@@ -0,0 +1,172 @@
+
+## How to find my current image tag
+
+The environment variable `JUPYTER_IMAGE_SPEC` will tell you what Pangeo Docker Image is currently in use.
+
+### Using the command line with Bash
+
+```bash
+echo JUPYTER_IMAGE_SPEC: $JUPYTER_IMAGE_SPEC
+
+tmp_array=($(echo $JUPYTER_IMAGE_SPEC | tr "/:" "\n"))
+image_name=${tmp_array[1]}
+image_tag=${tmp_array[2]}
+
+echo image_name: $image_name
+echo image_tag: $image_tag
+```
+
+### Using a Jupyter notebook with Python
+
+```python
+import os
+image_spec = os.environ['JUPYTER_IMAGE_SPEC']
+print('JUPYTER_IMAGE_SPEC:', image_spec)
+
+_, image_name, image_tag = image_spec.replace('/',':').split(':')
+
+print('image_name:', image_name)
+print('image_tag:', image_tag)
+```
+
+## How to find the list of available image tags:
+
+Using GitHub the web site, the list of select a particular tag from the branches/tag drop down menu.
+
+## How to finding list of packages in any given tag
+
+Each image has an `packages.txt` file that lists exactly which packages and versions installed in the image.
+
+### Using GitHub website
+
+Using GitHub the web [site](https://github.com/pangeo-data/pangeo-docker-images), select a particular tag from the branches/tag drop down menu.  Navigate to the image and select the `packages.txt` file.
+
+### Using the command line with Bash
+
+```bash
+tmp_array=($(echo $JUPYTER_IMAGE_SPEC | tr "/:" "\n"))
+image_name=${tmp_array[1]}
+image_tag=${tmp_array[2]}
+
+packages_url=https://raw.githubusercontent.com/pangeo-data/pangeo-docker-images/$image_tag/$image_name/packages.txt
+
+curl -s $packages_url
+```
+
+### Using a Jupyter notebook with Python
+
+```python
+import os
+image_spec = os.environ['JUPYTER_IMAGE_SPEC']
+_, image_name, image_tag = image_spec.replace('/',':').split(':')
+
+def get_package_list(image_name, image_tag):
+    """Return a list of packages with versions in an image"""
+    packages_url = f"https://raw.githubusercontent.com/pangeo-data/pangeo-docker-images/{image_tag}/{image_name}/packages.txt"
+    
+    import requests
+    packages_str = requests.get(packages_url).content.decode('utf8')
+    
+    packages = []
+    for package in packages_str.splitlines():
+        if package[0] == '#': continue # skip the header
+        package_name, package_version = package.split('==')
+        packages.append((package_name, package_version))
+
+    return packages
+
+packages = get_package_list(image_name, image_tag)
+print(f"{len(packages)} packages found in {image_spec}\n")
+for package in packages:
+    print(f"{package[0]}=={package[1]}")
+```
+
+## How to find difference in packages lists between any two tags
+
+Suppose you want to find out what the differences are between 
+
+`pangeo/pangeo-notebook:2023.06.07`
+
+and
+
+`pangeo/pangeo-notebook:2023.06.20`
+
+to determine which packages were changed, added, or removed?
+
+### Using the command line with Bash
```

Comment:
Instead of using bash or python, we should allow people to do this directly from the GitHub UI. https://docs.github.com/en/pull-requests/committing-changes-to-your-project/viewing-and-comparing-commits/comparing-commits#comparing-tags has info. 

You can do so by:

1. Finding the release for the tag, from https://github.com/pangeo-data/pangeo-docker-images/tags
2. Select the tag, so you get something like https://github.com/pangeo-data/pangeo-docker-images/tags
3. Use the 'compare' dropdown on the left to select another tag to compare it against
4. Look at the packages.txt of the appropriate image.

IMO this is easier and more useful than having to write code to figure this out.

Reply:
I've added instructions for using the GitHub UI